polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -1079,35 +1045,35 @@ void LoopInternalization::transform(FunctionOpInterface func,
                                     const FunctionKernelInfo &funcKernelInfo,
                                     const unsigned localMemoryRemaining,
                                     DataFlowSolver &solver) {
-  // Calculate the required local memory for all accesses in
-  // loopToSharedMemref to be promoted.
+  // Calculate the total amount of shared memory needed to promote the memory
+  // accesses that were identified by the analysis.
   SmallVector<gpu::GPUFuncOp> kernels;
   funcKernelInfo.getKernelCallers(func, kernels);
+
+  const unsigned numDims = getGridDimension(func);
```

Comment:
Can you please add back the assertion?
`assert(numDims > 0 && numDims <= 3 && "Dimension out of range");`

Reply:
`getGridDimension` contains the assertion already:

```
        unsigned dim = ty.getDimension();
        assert(dim > 0 && dim <= 3 && "Dimension out of range");
        return dim;

Reply:
done.

Reply:
`getGridDimension` may returns 0 if it cannot find the grid dimension, and we want to assert for that case.

Reply:
add this check in getCandidateFunction.